muskrat/s3consumer.py

```diff
@@ -125,12 +142,8 @@ def consume(self):
         #Update: actually... this doesn't seem to be a problem...
         msg_iterator = self._get_msg_iterator()
 
-        for msg in msg_iterator:
-            #Sub 'directories' are prefix objects, so ignore them
-            if isinstance( msg, boto.s3.key.Key ):
-                self.callback( msg.get_contents_as_string() )
-                self._cursor.update( msg.name )
-
+        for obj in self._cursor.each(msg_iterator):
+            self.callback(obj.get()['Body'].read())
```

Comment:
I just created a test for this. Just to be sure that I have it right:

Given that you are monitoring the prefix "FOO/BAR/".
And the bucket is empty.

When you add object A with key "FOO/BAR/2016-09-21T13:53:23.594894"
And you add object B with key "FOO/BAR/BAZ/2016-09-21T13:54:37.164853"
And the muskrat consumer is invoked

Then we should process object A
And we should not process object B

Is this the right test? If so, the implementation in this PR fails -- it processes both A and B.